api/v1alpha1/nonadmindownloadrequest_types.go

```diff
@@ -0,0 +1,117 @@
+	Phase NonAdminPhase `json:"phase,omitempty"`
+
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=nonadmindownloadrequests,shortName=nadr
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.phase"
+
+// NonAdminDownloadRequest is the Schema for the nonadmindownloadrequests API.
+type NonAdminDownloadRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NonAdminDownloadRequestSpec   `json:"spec,omitempty"`
+	Status NonAdminDownloadRequestStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// NonAdminDownloadRequestList contains a list of NonAdminDownloadRequest.
+type NonAdminDownloadRequestList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NonAdminDownloadRequest `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&NonAdminDownloadRequest{}, &NonAdminDownloadRequestList{})
+}
+
+// NonAdminDownloadRequestConditionType prevents untyped strings for NADR conditions functions
+type NonAdminDownloadRequestConditionType string
+
+const (
+	// ConditionNonAdminBackupStorageLocationNotUsed block download requests processing if NaBSL is not used
+	ConditionNonAdminBackupStorageLocationNotUsed NonAdminDownloadRequestConditionType = "NonAdminBackupStorageLocationNotUsed"
+	// ConditionNonAdminBackupNotAvailable indicates backup is not available, and will backoff download request
+	ConditionNonAdminBackupNotAvailable NonAdminDownloadRequestConditionType = "NonAdminBackupNotAvailable"
+	// ConditionNonAdminRestoreNotAvailable indicates restore is not available, and will backoff download request
+	ConditionNonAdminRestoreNotAvailable NonAdminDownloadRequestConditionType = "NonAdminRestoreNotAvailable"
+	// ConditionNonAdminProcessed indicates that the NADR is in a terminal state
+	ConditionNonAdminProcessed NonAdminDownloadRequestConditionType = "Processed"
+)
+
+// ReadyForProcessing returns if this NonAdminDownloadRequests is in a state ready for processing
+// only process NADR with target kind and name populated and without terminal condition.
+//
+// Terminal condition includes
+// - NonAdminBackupStorageLocationNotUsed: we currently require NaBSL usage on the NAB/NAR to process this download request
+//
+// returns true if ready for processing, false if required fields are not populated
+func (nadr *NonAdminDownloadRequest) ReadyForProcessing() bool {
+	// if nadr has ConditionNonAdminBackupStorageLocationUsed return false
+	if nadr.Status.Conditions != nil {
+		for _, condition := range nadr.Status.Conditions {
+			if condition.Type == string(ConditionNonAdminBackupStorageLocationNotUsed) &&
+				condition.Status == metav1.ConditionTrue {
+				return false
+			}
+		}
+	}
+	return nadr.Spec.Target.Kind != constant.EmptyString &&
```

Comment:
I think this does not make sense. Are not these fields required? it would always return true

Reply:
If I add +kubebuilder:validation:Required then I can replace this with a comment.

Reply:
no no, I think this is already done in velero side

Reply:
Do you want this check replaced with comment?

Reply:
just change return true, no need for comment